massa-execution-worker/src/active_history.rs

```diff
@@ -83,7 +83,9 @@ impl ActiveHistory {
                     return HistorySearchResult::Present(SetUpdateOrDelete::Set(msg));
                 }
                 Some(SetUpdateOrDelete::Update(msg_update)) => {
-                    current_updates.apply(msg_update.clone());
+                    let mut combined_message_update = msg_update.clone();
+                    combined_message_update.apply(current_updates);
+                    current_updates = combined_message_update;
                 }
                 Some(SetUpdateOrDelete::Delete) => return HistorySearchResult::Absent,
```

Comment:
how about this ? if the `current_updates` actually creates an element while it was deleted before... can it happen?